src/pages/MainQuizScreen.tsx

```diff
@@ -54,31 +73,95 @@ const isCorrectAnswer = (questionId: number, chosenAnswerId: number) => {
 };
 
 const MainQuizScreen: React.FC = () => {
-  const [currentQuestionIndex, setCurrentQuestionIndex] = useState(0);
+  const [currentQuestionId, setCurrentQuestionId] = useState<number>(1);
+  const [feedback, setFeedback] = useState<string>('');
+  const [isQuizCompleted, setIsQuizCompleted] = useState<boolean>(false);
+
+  const [question, answers]: [Question | undefined, Answer[]] = useMemo(() => {
+    const _question = quizQuestions.find((q) => q.id === currentQuestionId);
+    const _answers = getAnswersByQuestionId(currentQuestionId);
+    return [_question, _answers];
+  }, [currentQuestionId]);
 
-  const nextQuestion = () => {
-    setCurrentQuestionIndex((prevIndex) => prevIndex + 1);
-  }; // for the future implementation of changing questions;
+  useEffect(() => {
+    if (feedback) {
+      const timer = setTimeout(() => {
+        handleNextQuestion();
+        setFeedback('');
+      }, 2000);
+      return () => clearTimeout(timer);
+    }
+  }, [feedback]);
+
+  const handleAnswerClick = (chosenAnswerId: number) => {
+    if (isCorrectAnswer(currentQuestionId, chosenAnswerId)) {
+      setFeedback("It's correct!");
+    } else {
+      setFeedback("It's incorrect!");
+    }
```

Comment:
[nitpicking]
```ts
const FEEDBACK = {
  CORRECT: "It's correct!",
  INCORRECT: "It's incorrect!",
  default: ''
}
//...
setFeedback(() => isCorrectAnswer(currentQuestionId, chosenAnswerId) ? FEEDBACK.CORRECT : FEEDBACK.INCORRECT)
```
this is minor change, feel free to skip it in this PR and add in the following one :)